### PR TITLE
Handle findSourceMap returning null since Node.js 18.8.0

### PR DESCRIPTION
**Pull request by Brooooooklyn**

### CODE REVIEW

**Comment by iambumblehead** [comment]
this issue is causing the CI pipeline here to fail https://github.com/iambumblehead/esmock/runs/8011056406?check_suite_focus=true

File: lib/snapshot-manager.js [hunk 1]
```diff
@@ -391,7 +391,10 @@ class Manager {
 
 const resolveSourceFile = mem(file => {
 	const sourceMap = findSourceMap(file);
-	if (sourceMap === undefined) {
+	// since Node.js 18.8.0, the return value of `findSourceMap` changed to null if not found
+	// it was `undefined` in previous version
+	// https://github.com/nodejs/node/pull/43875
+	if (sourceMap == undefined) {
```
Comment (sindresorhus) [inline lib/snapshot-manager.js:397]:
```suggestion
	if (sourceMap === undefined || sourceMap === null) {
```

**Comment by novemberborn** [comment]
Nice find @Brooooooklyn. Too bad the documentation doesn't define behavior when the source map cannot be found. I guess that's what we get for using an experimental feature.

**Review by novemberborn — APPROVED** [review]

**Comment by novemberborn** [comment]
Should be fixed in 4.3.3.

**Pull request merged**